### PR TITLE
add replaceSelectionItems

### PR DESCRIPTION
**Pull request by ellell**
Type: Minor
Review: @kesla

### CODE REVIEW

File: lib/replace-selection-items.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {splice} from 'immutable-array-methods';
+import getSelectionItems from './get-selection-items';
+
+export default (articleJson, fn) => {
+  const items = getSelectionItems(articleJson);
+
+  if (!items.length) {
+    return articleJson;
+  }
```
Comment (kesla) [inline lib/replace-selection-items.js]: I'm pretty sure we handle this in `immutable-array-methods` - that doing splice w deleteCount === 0 & no items will be fine.

Comment (kesla) [inline lib/replace-selection-items.js]: and that we'll then return the original array

File: lib/replace-selection-items.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  const selectionStartIndex = articleJson.indexOf(items[0]);
+  const deleteCount = items.length;
+  return splice(articleJson, selectionStartIndex, deleteCount, fn(items));
```
Comment (kesla) [inline lib/replace-selection-items.js]: Don't we want to do `splice(articleJson, selectionStartIndex, deleteCount, ...fn(items));` here? Right now we can only replace it with a single element - so we can implement `wrap` with this but not `unwrap` - for example.

Or am I reading the code wrong? Let's implement a test that does `unwrap` to be sure that use case works!

File: lib/replace-selection-items.js (hunk 1 above)
Comment (kesla) [inline lib/replace-selection-items.js]: oh I see now - if we've got `items.length === 0` we don't want to call `fn(items)` further down. So this is indeed needed :)

File: lib/replace-selection-items.js (hunk 2 above)
Comment (ellell) [inline lib/replace-selection-items.js]: @kesla Fixed!

**Comment by kesla** [comment]
looks great!

**Pull request merged**